src/Microsoft.Health.Dicom.SqlServer/Features/Schema/Migrations/5.diff.sql

```diff
@@ -0,0 +1,1613 @@
+------------------------------------------------------------------------
+CREATE OR ALTER PROCEDURE dbo.AddInstance
+    @partitionId                        VARCHAR(64)    = 'Microsoft.Default',
+    @studyInstanceUid                   VARCHAR(64),
+    @seriesInstanceUid                  VARCHAR(64),
+    @sopInstanceUid                     VARCHAR(64),
+    @patientId                          NVARCHAR(64),
+    @patientName                        NVARCHAR(325) = NULL,
+    @referringPhysicianName             NVARCHAR(325) = NULL,
+    @studyDate                          DATE = NULL,
+    @studyDescription                   NVARCHAR(64) = NULL,
+    @accessionNumber                    NVARCHAR(64) = NULL,
+    @modality                           NVARCHAR(16) = NULL,
+    @performedProcedureStepStartDate    DATE = NULL,
+    @patientBirthDate                   DATE = NULL,
+    @manufacturerModelName              NVARCHAR(64) = NULL,
+    @stringExtendedQueryTags dbo.InsertStringExtendedQueryTagTableType_1 READONLY,
+    @longExtendedQueryTags dbo.InsertLongExtendedQueryTagTableType_1 READONLY,
+    @doubleExtendedQueryTags dbo.InsertDoubleExtendedQueryTagTableType_1 READONLY,
+    @dateTimeExtendedQueryTags dbo.InsertDateTimeExtendedQueryTagTableType_1 READONLY,
+    @personNameExtendedQueryTags dbo.InsertPersonNameExtendedQueryTagTableType_1 READONLY,
+    @initialStatus                      TINYINT
+AS
+    SET NOCOUNT ON
+
+    SET XACT_ABORT ON
+    BEGIN TRANSACTION
+
+    DECLARE @currentDate DATETIME2(7) = SYSUTCDATETIME()
+    DECLARE @existingStatus TINYINT
+    DECLARE @newWatermark BIGINT
+    DECLARE @studyKey BIGINT
+    DECLARE @seriesKey BIGINT
+    DECLARE @instanceKey BIGINT
+    DECLARE @partitionKey BIGINT
+
+    SELECT @partitionKey = PartitionKey
+    FROM dbo.Partition WITH(UPDLOCK)
+    WHERE PartitionId = @partitionId
+
+    IF @@ROWCOUNT = 0
+    BEGIN
+        SET @partitionKey = NEXT VALUE FOR dbo.PartitionKeySequence
+
+        INSERT INTO dbo.Partition
+            (PartitionKey, PartitionId, CreatedDate)
+        VALUES
+            (@partitionKey, @partitionId, SYSUTCDATETIME())
+    END
+
+    SELECT @existingStatus = Status
+    FROM dbo.Instance
+    WHERE PartitionKey = @partitionKey
+        AND StudyInstanceUid = @studyInstanceUid
+        AND SeriesInstanceUid = @seriesInstanceUid
+        AND SopInstanceUid = @sopInstanceUid
+
+    IF @@ROWCOUNT <> 0
+        -- The instance already exists. Set the state = @existingStatus to indicate what state it is in.
+        THROW 50409, 'Instance already exists', @existingStatus;
+
+    -- The instance does not exist, insert it.
+    SET @newWatermark = NEXT VALUE FOR dbo.WatermarkSequence
+    SET @instanceKey = NEXT VALUE FOR dbo.InstanceKeySequence
+
+    -- Insert Study
+    SELECT @studyKey = StudyKey
+    FROM dbo.Study WITH(UPDLOCK)
+    WHERE StudyInstanceUid = @studyInstanceUid
+    AND PartitionKey = @partitionKey
+
+    IF @@ROWCOUNT = 0
```

Comment:
It is very hard to review this file. I will see if 4.sql diff with 5.sql is better.